src/mcpdotnet/Server/McpServerExtensions.cs

```diff
@@ -1,45 +1,230 @@
 ﻿using McpDotNet.Protocol.Messages;
 using McpDotNet.Protocol.Types;
 using McpDotNet.Utils;
+using Microsoft.Extensions.AI;
+using System.Runtime.CompilerServices;
+using System.Text;
 
 namespace McpDotNet.Server;
 
 /// <inheritdoc />
 public static class McpServerExtensions
 {
     /// <summary>
-    /// Requests the client to create a new message.
+    /// Requests to sample an LLM via the client.
     /// </summary>
+    /// <exception cref="ArgumentNullException"><paramref name="server"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException">The client does not support sampling.</exception>
     public static Task<CreateMessageResult> RequestSamplingAsync(
         this IMcpServer server, CreateMessageRequestParams request, CancellationToken cancellationToken)
     {
         Throw.IfNull(server);
 
         if (server.ClientCapabilities?.Sampling is null)
         {
-            throw new McpServerException("Client does not support sampling");
+            throw new ArgumentException("Server's client does not support sampling.", nameof(server));
```

Comment:
I found the original message easy to understand, "server's client" took a bit of squinting to parse.